File: 卡码网/Python/test18.py
#18. 链表的基本操作
import sys

class Node():
	def __init__(self,value):
		self.value=value
		self.length=0
		self.next=None
	def show(self):
		if self.length==0:
			print("Link list is empty",end="")
		else:
			tmp=self.next
			while tmp!=None:
				if tmp!=self.next:
					print(" ",end="")
				print(tmp.value,end="")
				tmp=tmp.next
	def insert(self,index,value):
		if index<1 or index>self.length+1:
			return False
		node=Node(value)
		if index==1:
			node.next=self.next
			self.next=node
		else:
			i=1
			tmp=self.next
			while tmp!=None:
				i+=1
				if i==index:
					break
				tmp=tmp.next
			node.next=tmp.next
			tmp.next=node
		self.length+=1
		return True
	def delete(self,index):
		if index<1 or index>self.length:
			return False
		tmp=self.next
		if index==1:
			self.next=tmp.next
		else:
			i=1
			while tmp!=None:
				i+=1
				if i==index:
					break
				tmp=tmp.next
			tmp.next=tmp.next.next
		self.length-=1
		return True
	def get(self,index):
		if index<1 or index>self.length:
			return None
		tmp=self.next
		i=1
		while tmp!=None:
			if i==index:
				return tmp
			i+=1
			tmp=tmp.next

def main():
	nums=list(map(int,input().split()))
	root=Node(-1)
	for i in range(1,nums[0]+1):
		root.insert(1,nums[i])
	n=int(input())
	for i in range(n):
		line=input().split()
		if line[0]=="show":
			root.show()
		elif line[0]=="delete":
			index=int(line[1])
			flag=root.delete(index)
			if flag:
				print("delete OK",end="")
			else:
				print("delete fail",end="")
		elif line[0]=="get":
			index=int(line[1])
			node=root.get(index)
			if node==None:
				print("get fail",end="")
			else:
				print(node.value,end="")
		else:
			index=int(line[1])
			value=int(line[2])
			flag=root.insert(index,value)
			if flag:
				print("insert OK",end="")
			else:
				print("insert fail",end="")
		print()

if __name__ == '__main__':
	main()

File: 卡码网/Js/test18.js
//18. 链表的基本操作
const rl=require("readline").createInterface({input:process.stdin});
const iter=rl[Symbol.asyncIterator]();
const readline=async ()=>(await iter.next()).value;
const out=process.stdout;

class Node{
  constructor(value){
    this.data=value;
    this.length=0;
    this.next=null;
  }
  show(){
    if(this.length===0) out.write("Link list is empty");
    else{
      let tmp=this.next;
      while(tmp){
        if(tmp!=this.next) out.write(" ");
        out.write(tmp.data.toString());
        tmp=tmp.next;
      }
    }
  }
  insert(index,value){
    if(index<1||index>this.length+1) return false;
    let node=new Node(value);
    let i=1;
    if(i===index){
      node.next=this.next;
      this.next=node;
    }else{
      let tmp=this.next;
      while(tmp){
        i++;
        if(i==index) break;
        tmp=tmp.next;
      }
      node.next=tmp.next;
      tmp.next=node;
    }
    this.length++;
    return true;
  }
  deleteNode(index){
    if(index<1||index>this.length||this.length===0) return false;
    let tmp=this.next;
    let i=1;
    if(i===index) this.next=tmp.next;
    else{
      while(tmp){
        i++;
        if(i===index) break;
        tmp=tmp.next;
      }
      tmp.next=tmp.next.next;
    }
    this.length--;
    return true;
  }
  getNode(index){
    if(index<1||index>this.length) return null;
    let i=1;
    if(i===index) return this.next;
    let tmp=this.next;
    while(tmp){
      i++;
      tmp=tmp.next;
      if(i===index){
        return tmp;
      }
    }
  }
}

async function main(){
  const nums=(await readline()).split(" ").map(Number);
  let root=new Node(-1);
  for(let i=1;i<=nums[0];i++){
    root.insert(1,nums[i]);
  }
  const n=parseInt(await readline());
  let index;
  for(let i=0;i<n;i++){
    let line=(await readline()).split(" ");
    let op=line[0];
    let flag=false;
    switch(op){
      case "show":
        root.show();
        break;
      case "get":
        index=parseInt(line[1]);
        let node=root.getNode(index);
        if(node) out.write(node.data.toString());
        else out.write("get fail");
        break;
      case "delete":
        index=parseInt(line[1]);
        flag=root.deleteNode(index);
        if(flag) out.write("delete OK");
        else out.write("delete fail");
        break;
      case "insert":
        index=parseInt(line[1]);
        flag=root.insert(index,parseInt(line[2]));
        if(flag) out.write("insert OK");
        else out.write("insert fail");
        break;
    }
    console.log();
  }
}

main();